queries/sourcepawn/highlights.scm

```diff
@@ -0,0 +1,306 @@
+(identifier) @variable
+
+; Assume all-caps names are constants
+((identifier) @constant
+  (#lua-match? @constant "^[A-Z][A-Z0-9_]+$"))
+
+; Function definitions/declarations
+(function_definition
+  name: (identifier) @function)
+
+(function_declaration
+  name: (identifier) @function)
+
+(parameter_declaration
+  name: (identifier) @variable.parameter)
+
+; Methods / Properties
+(field_access
+  field: (identifier) @variable.member)
+
+; Function calls
+(call_expression
+  function: (identifier) @function)
+
+(call_expression
+  function:
+    (field_access
+      field: (identifier) @function.method.call))
+
+; Types
+[
+  (builtin_type)
+  (any_type)
+] @type.builtin
+
+(type
+  (identifier) @type)
+
+; Variables
+(variable_declaration
+  name: (identifier) @variable)
+
+(old_variable_declaration
+  name: (identifier) @variable)
+
+; Preprocessor
+[
+  (preproc_include)
+  (preproc_tryinclude)
+] @keyword.import
+
+[
+  (system_lib_string)
+  (string_literal)
+] @string
+
+(preproc_arg) @constant
+
+(preproc_macro) @function.macro
+
+(macro_param) @variable.parameter
+
+[
+  (preproc_assert)
+  (preproc_pragma)
+  (preproc_if)
+  (preproc_else)
+  (preproc_elseif)
+  (preproc_endif)
+  (preproc_endinput)
+  (preproc_error)
+  (preproc_warning)
+] @keyword.directive
+
+[
+  (preproc_define)
+  (preproc_undefine)
+] @keyword.directive.define
+
+(preproc_define
+  name: (identifier) @constant)
+
+(preproc_undefine
+  name: (identifier) @constant)
+
+; Expressions
+(view_as) @function.builtin
+
+(sizeof_expression) @function.macro
+
+[
+  (this)
+  ; https://github.com/alliedmodders/sourcemod/blob/5c0ae11a4619e9cba93478683c7737253ea93ba6/plugins/include/handles.inc#L78
+  (hardcoded_symbol)
+] @variable.builtin
+
+; Comments
+(comment) @comment
+
+; General
+(parameter_declaration
+  defaultValue: (identifier) @constant)
+
+[
+  (fixed_dimension)
+  (dimension)
+  (array_indexed_access)
+] @punctuation.bracket
```

Comment:
Do these include the index?

Reply:
Only `fixed_dimension`.

Reply:
Is it highlighted as a number below?

Reply:
Yep, just rechecked, it looks good in practice 👍